test/extensions/quic_listeners/quiche/integration/quic_protocol_integration_test.cc

```diff
@@ -2,8 +2,11 @@
 
 namespace Envoy {
 
-// We do not yet run QUIC downstream tests.
-GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST(DownstreamProtocolIntegrationTest);
+// TODO(danzh) this should run with QUIC downstream instead of HTTP2.
+INSTANTIATE_TEST_SUITE_P(Protocols, DownstreamProtocolIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams(
+                             {Http::CodecClient::Type::HTTP2}, {FakeHttpConnection::Type::HTTP3})),
```

Comment:
suggest to test HTTP and HTTP2 for downstream as well. I will add another test suite for downstream HTTP3.

Reply:
I find a test for h3 downstream work for H1 but not H2 upstream. So worth to run both I think.

Reply:
How about I land this as-is, and after yours is, see if adding the extras nets anything new?  I'm mildly skeptical it will and either way I think incremental change should be fine.

Reply:
SGTM